hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/testutils/SparkClientFunctionalTestHarness.java

```diff
@@ -312,9 +322,17 @@ protected HoodieWriteConfig getConfig(Boolean autoCommit, Boolean rollbackUsingM
         .withStorageConfig(HoodieStorageConfig.newBuilder().hfileMaxFileSize(1024 * 1024 * 1024).parquetMaxFileSize(1024 * 1024 * 1024).build())
         .withEmbeddedTimelineServerEnabled(true).forTable("test-trip-table")
         .withFileSystemViewConfig(new FileSystemViewStorageConfig.Builder()
+            .withRemoteServerPort(timelineServicePort)
             .withEnableBackupForRemoteFileSystemView(false).build())
         .withIndexConfig(HoodieIndexConfig.newBuilder().withIndexType(indexType).build())
         .withClusteringConfig(clusteringConfig)
         .withRollbackUsingMarkers(rollbackUsingMarkers);
   }
+
+  protected int incrementTimelineServicePortToUse() {
```

Comment:
but you can have ports be held by other processes within the system? Can we try and obtain free ports like we do with HDFSTestService?

Reply:
As discussed offline, the config passed here specifies the starting port to attempt to bind for the timeline server.  The explicit port rotation here is to avoid test failures due to successful port binding with connection reset later on.